src/core/fem/src/discretization/4C_fem_discretization_utils.cpp

```diff
@@ -121,4 +121,35 @@ void Core::FE::Utils::do_initial_field(const Core::Utils::FunctionManager& funct
   }
 }
 
+/*----------------------------------------------------------------------------*
+ *----------------------------------------------------------------------------*/
+Core::LinAlg::MultiVector<double> Core::FE::Utils::average_element_to_nodal_values(
+    const Discretization& discretization,
+    const Core::LinAlg::MultiVector<double>& element_multi_vector)
+{
```

Comment:
At the start you probably want to check some preconditions: are the vector sizes consistent with data in the discretization? Is the discretization even filled? etc.